roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -22,6 +27,26 @@
     HAVE_NWB = False
 
 
+def dict_recursive_update(metadata_base, metadata_input):
+    return_dict = deepcopy(metadata_base)
+    for base_key, base_val in return_dict.items():
+        if metadata_input.get(base_key):
+            if isinstance(base_val,dict):
+                return_dict[base_key] = dict_recursive_update(base_val, metadata_input[base_key])
+            elif isinstance(base_val,list):
+                if isinstance(metadata_input[base_key], list):
+                    for base_val_num,(a,b) in enumerate(zip(base_val, metadata_input[base_key])):
+                        if isinstance(a, dict) and isinstance(b, dict):
+                            return_dict[base_key][base_val_num] = dict_recursive_update(a, b)
+                        elif isinstance(a, (int, float, str)) and isinstance(b, (int, float, str)):
+                            return_dict[base_key][base_val_num] = b
+                else:
+                    continue
+            else:
+                return_dict[base_key] = metadata_input[base_key]
+    return return_dict
```

Comment:
```suggestion
def dict_recursive_update(base, input_):
    for key, val in input_.items():
        if isinstance(val, dict) and isinstance(base[key], dict):
            dict_recursive_update(base[key], val)
        elif isinstance(val, list) and isinstance(base[key], list):
            for i, input_list_item in enumerate(val):
                if len(base[key]) > i:
                    if isinstance(base[key][i], dict) and isinstance(input_list_item, dict):
                        dict_recursive_update(base[key][i], input_list_item)
                    else:
                        base[key][i] = input_list_item
                else:
                    base[key].append(input_list_item)
        else:
            base[key] = val
```

Reply:
I've put my comments as python comments
```python
def dict_recursive_update(base, input_):
    for key, val in input_.items():
        if isinstance(val, dict) and isinstance(base[key], dict): #this would throw an error if base[key] is not found
            dict_recursive_update(base[key], val)
        elif isinstance(val, list) and isinstance(base[key], list):
# I was wanting to define the length of the list with the predefined base dict. If for example, base data expects 
# 1 roiresponse series (since only 1 was found in SegExt object, but the user inputs the names for two then, 
# it would add the second one to the base, the data for which is non existant
            for i, input_list_item in enumerate(val):
                if len(base[key]) < i: 
                    if isinstance(base[key][i], dict) and isinstance(input_list_item, dict):
                        dict_recursive_update(base[key][i], input_list_item)
                    else:
                        base[key][i] = input_list_item
                else:
                    base[key].append(input_list_item)#input_ would then be expected to have all correct nwbfile specific fields, it would throw an error if the user inputs a wrong field.
        else:
            base[key] = val # if isinstance(base[key][i], dict) is true but input  isinstance(input_list_item, str/int) then it would be a problem
```

Reply:
I didn't understand all of your comments but I did fix one error you caught. How about this?

```python
def dict_recursive_update(base, input_):
    for key, val in input_.items():
        if key in base and isinstance(val, dict) and isinstance(base[key], dict):
            dict_recursive_update(base[key], val)
        elif key in base and isinstance(val, list) and isinstance(base[key], list):
            for i, input_list_item in enumerate(val):
                if len(base[key]) < i:
                    if isinstance(base[key][i], dict) and isinstance(input_list_item, dict):
                        dict_recursive_update(base[key][i], input_list_item)
                    else:
                        base[key][i] = input_list_item
                else:
                    base[key].append(input_list_item)
        else:
            base[key] = val
```

Reply:
okay this works, i've picked this method in the new commit